todoman/interactive.py

```diff
@@ -0,0 +1,583 @@
+        '''
+        Open a page over the existing stack of pages. pageToOpen is the
+        TodomanPage object to display.
+
+        (TodomanPage, TodomanPage) -> None
+        '''
+        self.parent._open_page(pageToOpen)
+
+    @property
+    def statusbar(self):
+        '''
+        Returns the current contents of the statusbar of the Todomanpage.
+
+        (TodomanPage) -> str
+        '''
+        return self.footer.contents[0].original_widget.text
+
+    @statusbar.setter
+    def statusbar(self, text):
+        '''
+        Sets the given text as the current text in the statusbar of the
+        TodomanPage.
+
+        (self, str) -> None
+        '''
+        self.footer.contents
+        self.footer.contents[0][0].original_widget.set_text(text)
+
+    def callback(self, **kwargs):
+        '''
+        A default callback function to use when closing the previous page. 
+        This callback function handles several keywords that are generic to
+        all callbacks. If a certain keyword is not set, the method does
+        nothing with it.
+
+        This callback function supports the following keywords:
+
+        statusbar (str): A text to set as the statusbar message
+
+        (TodomanPage, **kwargs) -> None
+        '''
+        for key, value in kwargs.items():
+            if key == 'statusbar':
+                self.statusbar = value
+        self.reload()
+
+    def reload(self):
+        '''
+        Dummy method. TodomanPage subclasses that need a reload should
+        implement one under this name.
+
+        (TodomanPage) -> None
+        '''
+        return None
+
+class TodomanItemListPage(TodomanPage):
+    '''
+    Class to contain a ListBox filled with todo items, based on a given
+    Database.
+    '''
+
+    def __init__(self, parent, callback, database):
+        '''
+        Create an instance of TodomanItemListPage. The parent is the
+        TodomanInteractive instance in which the page was created.
+        The database is the Database from which to display items.
+
+        Callback is the function to call when the page is closed. This function
+        is responsible for passing information to the underlying TodomanPage.
+        It is usually a method of the underlying TodomanPage.
+        
+        (TodomanItemListPage, TodomanInteractive, function, Database) -> None
+        '''
+        self.parent = parent
+        self.database = database
+        # By default, we hide the completed items
+        self.done_is_hidden = True 
+        items = self.items_to_display()
+        self.body = urwid.ListBox(urwid.SimpleFocusListWalker(items))
+        super().__init__(parent, callback)
+
+    def items_to_display(self):
+        '''
+        Create a list of TodomanItems to display, based on the associated
+        Database and the current done_is_hidden setting.
+
+        (TodomanItemListPage) -> [TodomanItem]
+        '''
+        items = []
+        for filename in self.database.todos.keys(): 
```

Comment:
If `TodomanItem` accepts a `todo` instead of a `filename`, you can just iterate through values here and save one key lookup in `TodomanItem.__init__`